cmd/src/search.go

```diff
@@ -440,6 +466,27 @@ func applyHighlights(input string, highlights []highlight, start, end string) st
 	return string(result)
 }
 
+// convertMatchToHighlights converts a FileMatch m to a highlight data type.
+// When isPreview is true, it is assumed that the result to highlight is only on
+// one line, and the offets are relative to this line. When isPreview is false,
+// the lineNumber from the FileMatch data is used, which is relative to the file
+// content.
+func convertMatchToHighlights(m map[string]interface{}, isPreview bool) (highlights []highlight) {
```

Comment:
I'm using `isPreview` to control whether we assume the line to show is 1, or whether to look at the match-reported line to extract it from file.